homeassistant/components/ohme/coordinator.py

```diff
@@ -0,0 +1,71 @@
+"""Ohme coordinators."""
+
+from datetime import timedelta
+import logging
+
+from ohme import ApiException, AuthException, OhmeApiClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+type OhmeConfigEntry = ConfigEntry[OhmeCoordinator]
+
+
+class OhmeCoordinator(DataUpdateCoordinator[None]):
+    """Coordinator to pull all updates from the API."""
+
+    config_entry: OhmeConfigEntry
+
+    def __init__(self, hass: HomeAssistant, entry: OhmeConfigEntry) -> None:
+        """Initialise coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="Ohme Coordinator",
+            update_interval=timedelta(seconds=30),
+            config_entry=entry,
+        )
+
+        self.client: OhmeApiClient = OhmeApiClient(
+            self.config_entry.data[CONF_EMAIL], self.config_entry.data[CONF_PASSWORD]
+        )
+        self._alternative_iteration: bool = True
+
+    async def _async_setup(self) -> None:
+        try:
+            await self.client.async_login()
+
+            if not await self.client.async_update_device_info():
+                raise ConfigEntryNotReady(
+                    translation_key="device_info_failed", translation_domain=DOMAIN
+                )
+        except AuthException as e:
+            raise ConfigEntryError(
+                translation_key="auth_failed", translation_domain=DOMAIN
+            ) from e
+        except ApiException as e:
+            raise ConfigEntryNotReady(
```

Comment:
```suggestion
            raise UpdateFailed(
```

Reply:
Will UpdateFailed also cause the integration setup to fail? Because if this call fails and the entities get setup anyway, the device info will be all messed up

Reply:
Yep, this is caught and rethrown as ConfigEntryNotReady